### PR TITLE
How to make a copy of the Publishing API database on your local machine

### PR DESCRIPTION
**Pull request by jonathanglassman**
## What

[Document](https://docs.data-community.publishing.service.gov.uk/analysis/access-publishing/#access-the-gov-uk-publishing-database) how to make a copy of the Publishing API database on your laptop, from a nightly backup on S3.

[Notes](https://docs.google.com/document/d/1vjIEz1CPo5ZHS3N1kwt110cpNq6J8vNbVnqGICxY81k/edit?pli=1#bookmark=id.vn6hutdxvly5)

Trello card: https://trello.com/c/2Yp468jJ/1506-document-using-the-publishing-database-backup

### CODE REVIEW

File: source/analysis/access-publishing/index.html.md.erb [hunk 1]
```diff
@@ -41,3 +43,57 @@ See the [Get started on GOV.UK developer documentation](https://docs.publishing.
...
     This database is named `production`, but it is not in the production environment.
+
+## Download a backup of the publishing database
+
+You can download a backup of the publishing database to your local machine instead of connecting to the integration environment database.
+
+You may want to do this if you want to:
+
+- run resource-intensive queries on the database, as these queries may affect live services that also use the database
+- create your own tables within the database, as you only have read access to the integration environment database
```
Comment (nacnudus) [inline source/analysis/access-publishing/index.html.md.erb:54]: Would this be worth repeating at line 16?

File: source/analysis/access-publishing/index.html.md.erb [hunk 2]
```diff
@@ -41,3 +43,57 @@ See the [Get started on GOV.UK developer documentation](https://docs.publishing.
...
+Before you start, you must:
+
+- have access to AWS
+- install the GDS command line tools
+
+You do not need the GDS VPN or Secure Shell (SSH) access to the integration environment.
+
+### Download the database backup
+
+1. The filename of the latest database backup in AWS S3 varies depending on the date and time the backup was created. Run the following in the command line to find out the filename:
```
Comment (nacnudus) [inline source/analysis/access-publishing/index.html.md.erb:65]: Is there a standard GDS way of referring to the command line or the terminal?  I would write either "at the command line" or "in the terminal", but I don't know why, and I'd bow to a convention.

File: source/analysis/access-publishing/index.html.md.erb [hunk 3]
```diff
@@ -41,3 +43,57 @@ See the [Get started on GOV.UK developer documentation](https://docs.publishing.
...
+    gds aws govuk-integration-readonly aws s3 ls s3://govuk-integration-database-backups/publishing-api-postgres/
+    ```
+
+1. Download the backup of the publishing api database:
+
+    ```sh
+    gds aws govuk-integration-readonly aws s3 cp s3://govuk-integration-database-backups/publishing-api-postgres/<YYYY-MM-DD>T<HH:MM:SS>-publishing_api_production.gz <DOWNLOAD-LOCATION>
+    ```
+    where:
+    - `<YYYY-MM-DD>T<HH:MM:SS>` is the date and time the backup was created.
```
Comment (nacnudus) [inline source/analysis/access-publishing/index.html.md.erb:77]:
```suggestion
    gds aws govuk-integration-readonly aws s3 cp s3://govuk-integration-database-backups/publishing-api-postgres/<FILENAME> <DOWNLOAD-LOCATION>
    ```
    where:
    - `<FILENAME>` is one of the filenames listed in the previous step.
```

File: source/analysis/access-publishing/index.html.md.erb [hunk 4]
```diff
@@ -41,3 +43,57 @@ See the [Get started on GOV.UK developer documentation](https://docs.publishing.
...
+    - `<DOWNLOAD-LOCATION>` is the location you want to download the backup file to on your local machine
```
Comment (nacnudus) [inline source/analysis/access-publishing/index.html.md.erb:78]: Nitpicking, but should this line end in a full stop, for consistency?

**Review by nacnudus — APPROVED** [review]
Thank you for this.  It is very clear, and the postgres section in particular is a vast improvement on the original notes.

A few questions and suggestions, but nothing to delay merging, so I am approving the PR.

File: source/analysis/access-publishing/index.html.md.erb (hunk 1 above)
Comment (jonathanglassman) [reply source/analysis/access-publishing/index.html.md.erb:54]: agree

File: source/analysis/access-publishing/index.html.md.erb (hunk 2 above)
Comment (jonathanglassman) [reply source/analysis/access-publishing/index.html.md.erb:65]: the command line is more standard and generic as users may not be using the Mac terminal, but some other command line interface

File: source/analysis/access-publishing/index.html.md.erb (hunk 4 above)
Comment (jonathanglassman) [reply source/analysis/access-publishing/index.html.md.erb:78]: whoops, i included a full stop in line 77 by mistake, good spot

**Pull request merged**